### ISSUE TITLE
parsing `D` gives `Maximum call stack size exceeded` error

### ISSUE THREAD

**Issue by strickinato** [cortex-js/compute-engine#32]
👋 Hello! Just bumped into an issue

Parsing `'D'` causes an infinite loop somewhere -- Here's a minimal example:

```js
> const ls = require('@cortex-js/compute-engine');
> const latexSyntax = new ls.LatexSyntax();
> latexSyntax.parse('d');
'd'

> latexSyntax.parse('D');
Uncaught RangeError: Maximum call stack size exceeded
    at l (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:1:3816)
    at u (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:1:4004)
    at ni.latexAhead (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:68001)
    at ni.lookAhead (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:68251)
    at ni.peekDefinition (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:68621)
    at ni.matchDefinition (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:72131)
    at ni.matchSymbol (/Users/aaronstrick/work/brilliant/node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:72311)
    at ni.matchPrimary (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:78836)
    at ni.matchExpression (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:79320)
    at ni.matchExpression (.../node_modules/@cortex-js/compute-engine/dist/compute-engine.min.js:7:79556)

```

**Occurred on**
Node version: `v14.17.3`
@cortex-js/compute-engine version: `0.4.2`

**Comment by strickinato** [comment]
In case it's helpful:

```js
letters.forEach(letter, l => {
  try {
    console.log(l, ': ', latexSyntax.parse(l))
  } catch {
    console.log(l, ': FAILED')
  }
})
```

Shows `D` is the only problematic character (at least in charcode 65-> 122 range)
```
A :  A
B :  B
C :  C
D : FAILED
E :  E
F :  F
G :  G
H :  H
I :  I
J :  J
K :  K
L :  L
M :  M
N :  N
O :  O
P :  P
Q :  Q
R :  R
S :  S
T :  T
U :  U
V :  V
W :  W
X :  X
Y :  Y
Z :  Z
[ :
\ :  \
] :
^ :  [ 'Power', 'Missing', 'Missing' ]
_ :  [ 'Subscript', 'Missing' ]
` :
a :  a
b :  b
c :  c
d :  d
e :  ExponentialE
f :  f
g :  g
h :  h
i :  ImaginaryUnit
j :  j
k :  k
l :  l
m :  m
n :  n
o :  o
p :  p
q :  q
r :  r
s :  s
t :  t
u :  u
v :  v
w :  w
x :  x
y :  y
z :  z

**Comment by strickinato** [comment]
This appears fixed in 0.4.3 and can be closed 😄
